### PR TITLE
Add CLI tests

### PR DESCRIPTION
**Pull request by dbarbuzzi**
The primary goal of this PR is to add CLI tests to the repo (migrating them from the test-automation repo). To that effect:

* The Makefile has been updated to use the `TARGETS` argument with consistency related to other projects (sparseml, sparsezoo) in this fashion of multiple different test suite targets
  * `make test` runs unit tests alone
  * `make test TARGETS=cli` runs unit tests and the newly added CLI tests
  * `make test TARGETS=examples` runs the unit tests and the preexisting example tests
    * This removes the (broken) `make test-examples` command and splits some of the existing tests which ran against Python files in the `examples` folder out of the “unit” tests
  * `make test TARGETS=cli,examples` can be used to run **_all_** tests
* Existing CLI tests are added to the repo
  * They can be executed with the aforementioned `TARGETS=cli`
  * An additional dev dependency is also added, `ndjson`, used to parse some of the output for validation (the current version, 0.3.1, has no additional dependencies).

### CODE REVIEW

**Review by dhuangnm — APPROVED** [review]
Thansk Domenic!

**Review by mgoin — APPROVED** [review]
LGTM!

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,18 @@ DOCDIR := docs
 MDCHECKGLOBS := 'docs/**/*.md' 'docs/**/*.rst' 'examples/**/*.md' 'scripts/**/*.md'
 MDCHECKFILES := CODE_OF_CONDUCT.md CONTRIBUTING.md DEVELOPING.md README.md
 SPARSEZOO_TEST_MODE := "true"
+
+TARGETS := ""  # targets for running pytests: cli,examples
 PYTEST_ARGS ?= ""
+ifneq ($(findstring cli,$(TARGETS)),cli)
+	PYTEST_ARGS := $(PYTEST_ARGS) --ignore tests/test_benchmark.py \
```
Comment (mgoin) [inline Makefile:15]:
```suggestion
	PYTEST_ARGS += --ignore tests/test_benchmark.py \
```

File: Makefile [hunk 2]
```diff
@@ -8,7 +8,18 @@ DOCDIR := docs
...
+	--ignore tests/test_check_hardware.py \
+	--ignore tests/test_run_inference.py \
+	--ignore tests/test_server.py
+endif
+ifneq ($(findstring examples,$(TARGETS)),examples)
+	PYTEST_ARGS := $(PYTEST_ARGS) --ignore tests/test_examples.py
```
Comment (mgoin) [inline Makefile:21]:
```suggestion
	PYTEST_ARGS += --ignore tests/test_examples.py
```

Comment (dbarbuzzi) [reply Makefile:21]: Will resolve this when migrating the examples tests.

**Pull request merged**